File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def solution(n, t, m, p):
    answer = ''
    numbers = ""
    num = 0
    digits = "0123456789ABCDEF"

    while len(numbers) < t * m:
        if num == 0: # 0 일때
            numbers += "0"
        else:
            temp = ""
            x = num

            while x > 0:
                temp = digits[x % n] + temp
                x //= n

            numbers += temp

        num += 1

    for i in range(p - 1, t * m, m): # t * m 까지인데 m 간격으로
        answer += numbers[i]

        if len(answer) == t:
            break
    return answer

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.cpp
#include <string>
using namespace std;

string solution(int n, int t, int m, int p) {
    string answer = "";
    string numbers = "";
    int num = 0;

    string digits = "0123456789ABCDEF";

    while (numbers.length() < t * m) {
        if (num == 0) {
            numbers += "0";
        }
        else {
            string temp = "";
            int x = num;

            while (x > 0) {
                temp = digits[x % n] + temp;
                x /= n;
            }

            numbers += temp;
        }

        num++;
    }

    for (int i = p - 1; i < t * m; i += m) {
        answer += numbers[i];

        if (answer.length() == t) {
            break;
        }
    }

    return answer;
}
